### PR TITLE
allow validation of fields nested inside array

### PR DESCRIPTION
**Pull request by lyziane**
This is for issue #9023

### CODE REVIEW

File: src/View/Form/EntityContext.php [hunk 1]
```diff
@@ -584,6 +584,11 @@ public function error($field)
             return $entity->errors(array_pop($parts));
         }
 
+        if (is_array($entity)) {
+            $entity = $this->entity(array_shift($parts));
+            return $entity->errors($field);
```
[the remark this answers is not in the document]
Comment (lyziane) [reply src/View/Form/EntityContext.php]: line 584 didn't have a blank line before so I just copied the style

File: src/Validation/Validator.php [hunk 2]
```diff
@@ -104,6 +105,8 @@ public function errors(array $data, $newRecord = true)
     {
         $errors = [];
 
+        $data = Hash::flatten($data);
```
Comment (markstory) [inline src/Validation/Validator.php]: How do you see this interacting with the existing [nested validator](https://book.cakephp.org/3.0/en/core-libraries/validation.html#nesting-validators) features?

**Review by markstory — CHANGES_REQUESTED** [review]
New features require tests to be merged. It also looks like you have caused existing tests to fail.

**Comment by lyziane** [comment]
I was happy a little too fast. It worked in my project...and then more issues. This is no good...

File: src/Validation/Validator.php (hunk 2 above)
Comment (lyziane) [reply src/Validation/Validator.php]: What I wanted is pretty much to store a modelless form inside my entity.

**Pull request closed without merging**